txgossip/priority.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txgossip
+
+import (
+	"slices"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/txpool"
+)
+
+// A LazyTransaction couples a [txpool.LazyTransaction] with its sender.
+type LazyTransaction struct {
+	*txpool.LazyTransaction
+	Sender common.Address
+}
+
+// TransactionsByPriority calls [txpool.TxPool.Pending] with the given filter,
+// collapses the results into a slice, and sorts said slice by decreasing gas
+// tip then chronologically. Transactions from the same sender are merely sorted
+// by increasing nonce.
+func (s *Set) TransactionsByPriority(filter txpool.PendingFilter) []*LazyTransaction {
```

Comment:
I assume the plan is for this to be used by the block builder. fwiw the normal block builder includes an optimization to skip an entire account in the case that one of the txs is not valid. I'm not sure how important of an optimization this is, and it is also something we can add later as needed... So I don't think it's worth changing for now, but maybe we should add a TODO to investigate it?